open-machine-learning-jupyter-book/deep-learning/image-segmentation.md

```diff
@@ -0,0 +1,973 @@
+
+        x = self._conv_relu(c5, 4096, 7)
+        x = layers.Dropout(rate=0.5)(x)
+        x = self._conv_relu(x, 4096, 1)
+        x = layers.Dropout(rate=0.5)(x)
+
+        x = layers.Conv2D(num_classes, 1, kernel_initializer='he_normal')(x)
+        x = layers.Conv2DTranspose(num_classes, 4,
+                                   strides=2,
+                                   padding='same',
+                                   kernel_initializer='he_normal')(x)
+        c4 = layers.Conv2D(num_classes, 1, kernel_initializer='he_normal')(c4)
+        x = layers.Add()([x, c4])
+
+        x = layers.Conv2DTranspose(num_classes, 32,
+                                   strides=16,
+                                   padding='same',
+                                   kernel_initializer='he_normal')(x)
+
+        outputs = x
+        return models.Model(inputs, outputs, name=self.version)
+
+    def _fcn_8s(self, inputs):
+        num_classes = self.num_classes
+
+        if self.base_model in ['VGG16',
+                               'VGG19',
+                               'ResNet50',
+                               'ResNet101',
+                               'ResNet152',
+                               'MobileNetV1',
+                               'MobileNetV2']:
+            c3, c4, c5 = self.encoder(inputs, output_stages=['c3', 'c4', 'c5'])
+        else:
+            c3, c4, c5 = self.encoder(inputs, output_stages=['c2', 'c3', 'c5'])
+
+        x = self._conv_relu(c5, 4096, 7)
+        x = layers.Dropout(rate=0.5)(x)
+        x = self._conv_relu(x, 4096, 1)
+        x = layers.Dropout(rate=0.5)(x)
+
+        x = layers.Conv2D(num_classes, 1, kernel_initializer='he_normal')(x)
+        x = layers.Conv2DTranspose(num_classes, 4,
+                                   strides=2,
+                                   padding='same',
+                                   kernel_initializer='he_normal')(x)
+        c4 = layers.Conv2D(num_classes, 1)(c4)
+        x = layers.Add()([x, c4])
+
+        x = layers.Conv2DTranspose(num_classes, 4,
+                                   strides=2,
+                                   padding='same',
+                                   kernel_initializer='he_normal')(x)
+        c3 = layers.Conv2D(num_classes, 1)(c3)
+        x = layers.Add()([x, c3])
+
+        x = layers.Conv2DTranspose(num_classes, 16,
+                                   strides=8,
+                                   padding='same',
+                                   kernel_initializer='he_normal')(x)
+
+        outputs = x
+        return models.Model(inputs, outputs, name=self.version)
+```
+
+### SegNet
+
+SegNet is a novel and practical deep fully convolutional neural network architecture for semantic pixel-wise segmentation. This core trainable segmentation engine consists of an encoder network, a corresponding decoder network followed by a pixel-wise classification layer. Here is the structure of SegNet:
```

Comment:
add a ref/citation for SegNet.

Same for other models (e.g. U-Net)

Reply:
I am not sure what I need to do since SegNet was added to the reference.bib part. Should I add it to the acknowledge part?

Reply:
Done.